pkg/asset/asset.go

```diff
@@ -55,3 +56,39 @@ func PersistToFile(asset WritableAsset, directory string) error {
 	}
 	return nil
 }
+
+// DeleteAssetFromDisk removes all the files for asset from disk.
+func DeleteAssetFromDisk(asset WritableAsset, directory string) error {
+	for _, f := range asset.Files() {
+		path := filepath.Join(directory, f.Filename)
+		if err := os.Remove(path); err != nil && !os.IsNotExist(err) {
+			return errors.Wrap(err, "failed to remove file")
+		}
+
+		dir := filepath.Dir(path)
+		ok, err := isDirEmpty(dir)
```

Comment:
Checking empty-ness and then acting upon it in a later syscall is racy.  Ideally, we'd just attempt the removal and silently ignore [`EEXIST` or `ENOTEMPTY`][1].  But I dunno what Windows does, and this is a tiny thing to go through the hoops to make this portable.  Do we care at the moment?  Is it worth filing an issue about fixing this at some later point?

[1]: http://pubs.opengroup.org/onlinepubs/009695399/functions/rmdir.html

Reply:
> Checking empty-ness and then acting upon it in a later syscall is racy.
couldn't find IsNotEmpty like IsNotExists ? Is there one?

Reply:
> Checking empty-ness and then acting upon it in a later syscall is racy.

i think its not going to be racy in this context as our store is not multiple call safe anyways. i'll add a comment.